File: backend/src/main/java/com/medisage/utils/ParamCheck.java
package com.medisage.utils;

public class ParamCheck {
    public static boolean isNullOrEmpty(String param){
        return param == null || param.trim().isEmpty();
    }

    public static boolean isAvailEmail(String email){
        String emailRegex = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$";
        return email != null && email.matches(emailRegex);
    }

    public static boolean isAbailPassword(String password){
        String LargeLetterRegex = ".*[A-Z].*";
        String digitRegex = ".*\\d.*";
        String specialCharRegex = ".*[!@#$%^&*(),.?:{}|<>].*";
        //至少包含一个大写字母、一个数字、一个特殊字符
        if(!password.matches(LargeLetterRegex)){
            return false;
        }
        if(!password.matches(digitRegex)){
            return false;
        }
        if(!password.matches(specialCharRegex)){
            return false;
        }
        return password != null && password.length() >= 8;//长度大于8
    }

    public static boolean isAbailUsername(String username){
        String usernameRegex = "^\\w+$";
        if (username == null || !username.matches(usernameRegex)) {
            return false;
        }
        return username.length() >= 3 && username.length() <= 30 && username.matches(usernameRegex);//长度3-30
    }

    public static boolean isAbailCode(String code){
        String codeRegex = "^\\d{6}$";
        return code != null && code.matches(codeRegex);
    }
}

File: frontend/src/lib/paramCheck.js
export const isNullOrEmpty = (param) => {
  return param == null || String(param).trim().length === 0
}

export const isAvailEmail = (email) => {
  const s = email == null ? '' : String(email)
  const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/
  return emailRegex.test(s)
}

export const isAbailPassword = (password) => {
  const s = password == null ? '' : String(password)

  const largeLetterRegex = /[A-Z]/
  const digitRegex = /\d/
  const specialCharRegex = /[!@#$%^&*(),.?:{}|<>]/

  if (!largeLetterRegex.test(s)) return false
  if (!digitRegex.test(s)) return false
  if (!specialCharRegex.test(s)) return false

  return s.length >= 8
}

export const isAbailUsername = (username) => {
  const s = username == null ? '' : String(username)
  const usernameRegex = /^\w+$/
  if (!usernameRegex.test(s)) return false
  return s.length >= 3 && s.length <= 30
}

export const isAvailVerifyCode = (code) => {
  const s = code == null ? '' : String(code).trim()
  return /^\d{6}$/.test(s)
}
